rostime/src/time.cpp

```diff
@@ -167,6 +167,29 @@ namespace ros
     nsec = nsec_sum;
 #endif
   }
+
+  void ros_monotonictime(uint32_t& sec, uint32_t& nsec)
+#ifndef WIN32
+    throw(NoHighPerformanceTimersException)
+#endif
+  {
+#ifndef WIN32
+#if HAS_CLOCK_GETTIME
+    timespec start;
+    clock_gettime(CLOCK_MONOTONIC, &start);
+    sec  = start.tv_sec;
+    nsec = start.tv_nsec;
+#else
+    // what to do if clock_gettime is not available???
+    struct timeval timeofday;
+    gettimeofday(&timeofday,NULL);
+    sec  = timeofday.tv_sec;
+    nsec = timeofday.tv_usec * 1000;
```

Comment:
It is possible to avoid this on macOS (also other BSD systems which don't have `clock_gettime`), but I'm hesitant to ask for it since the existing wall time system doesn't use it.

I've implemented something that is monotonic and works on Linux, and macOS for ROS 2:

https://github.com/ros2/rcl/blob/545cb5602c4bf9d000f0ebe62b9d95cb9bf59469/rcl/src/rcl/time_unix.c#L73-L102

There's also something for Windows:

https://github.com/ros2/rcl/blob/66208db876bba344a6733f4aa33451a028085612/rcl/src/rcl/time_win32.c#L49-L65

You're welcome to steal those implementations if you like.